private/bufpkg/bufmodule/bufmoduleref/module_reference.go

```diff
@@ -94,6 +95,21 @@ func (m *moduleReference) IdentityString() string {
 	return m.remote + "/" + m.owner + "/" + m.repository
 }
 
+// DiscoverRemote returns the first remote in a list of dependencies that is not "buf.build", if
+// any. If not, it defaults to "buf.build". It skips any nil reference. It's useful for targeting
+// right single-tenant BSR addresses.
+func DiscoverRemote(refs []ModuleReference) string {
+	for _, ref := range refs {
+		if ref == nil {
+			continue
+		}
+		if ref.Remote() != bufconnect.DefaultRemote {
+			return ref.Remote()
+		}
+	}
+	return bufconnect.DefaultRemote
```

Comment:
Do we want to infer here if there are multiple non-default remotes found?

Reply:
Since the BSR doesn't support federation anymore, and we're finishing migrating a few customers that still have it enabled, I don't see a scenario in which we'd ever have more than one non-default remote. 99% of the cases they'll all have the same remote, either the public BSR, or their private remote address. In the remaining cases, there will be just two, public and theirs, we want to choose the latter.